gax/src/main/java/com/google/api/gax/retrying/BasicResultRetryAlgorithm.java

```diff
@@ -29,37 +29,59 @@
  */
 package com.google.api.gax.retrying;
 
+import java.util.concurrent.CancellationException;
+
 /**
  * A basic implementation of {@link ResultRetryAlgorithm}. Using this implementation would mean that
  * all exceptions should be retried, all responses should be accepted (including {@code null}) and
  * no retrying process should ever be canceled.
  *
  * @param <ResponseT> attempt response type
  */
-public class BasicResultRetryAlgorithm<ResponseT> implements ResultRetryAlgorithm<ResponseT> {
+public class BasicResultRetryAlgorithm<ResponseT>
+    implements ResultRetryAlgorithmWithContext<ResponseT> {
   /**
    * Always returns null, indicating that this algorithm does not provide any specific settings for
    * the next attempt.
-   *
-   * @param prevThrowable exception thrown by the previous attempt ({@code null}, if none)
-   * @param prevResponse response returned by the previous attempt
-   * @param prevSettings previous attempt settings
    */
   @Override
   public TimedAttemptSettings createNextAttempt(
-      Throwable prevThrowable, ResponseT prevResponse, TimedAttemptSettings prevSettings) {
+      Throwable previousThrowable,
+      ResponseT previousResponse,
+      TimedAttemptSettings previousSettings) {
     return null;
   }
 
   /**
-   * Returns {@code true} if an exception was thrown ({@code prevThrowable != null}), {@code false}
-   * otherwise.
-   *
-   * @param prevThrowable exception thrown by the previous attempt ({@code null}, if none)
-   * @param prevResponse response returned by the previous attempt
+   * Always returns null, indicating that this algorithm does not provide any specific settings for
+   * the next attempt.
+   */
+  @Override
+  public TimedAttemptSettings createNextAttempt(
+      RetryingContext context,
+      Throwable previousThrowable,
+      ResponseT previousResponse,
+      TimedAttemptSettings previousSettings) {
+    return null;
```

Comment:
can you call `createNextAttempt()` (no context one) here for consistency with the `shouldRetry` implementation?